src/components/editor-page/editor-pane/upload-handler.ts

```diff
@@ -8,51 +8,29 @@ import { Editor } from 'codemirror'
 import i18n from 'i18next'
 import { uploadFile } from '../../../api/media'
 import { store } from '../../../redux'
-import { supportedMimeTypes } from './tool-bar/utils/upload-image-mimetypes'
+import { supportedMimeTypes } from '../../common/upload-image-mimetypes'
 
 export const handleUpload = (file: File, editor: Editor): void => {
   if (!file) {
     return
   }
-  const mimeType = file.type
-  if (!supportedMimeTypes.includes(mimeType)) {
+  if (!supportedMimeTypes.includes(file.type)) {
     // this mimetype is not supported
     return
   }
   const cursor = editor.getCursor()
   const uploadPlaceholder = `![${i18n.t('editor.upload.uploadFile', { fileName: file.name })}]()`
   const noteId = store.getState().noteDetails.id
+  const insertCode = (replacement: string) => {
+    editor.replaceRange(replacement, cursor, { line: cursor.line, ch: cursor.ch + uploadPlaceholder.length }, '+input')
+  }
   editor.replaceRange(uploadPlaceholder, cursor, cursor, '+input')
-  uploadFile(noteId, mimeType, file)
+  uploadFile(noteId, file)
     .then(({ link }) => {
-      editor.replaceRange(
-        getCorrectSyntaxForLink(mimeType, link),
-        cursor,
-        {
-          line: cursor.line,
-          ch: cursor.ch + uploadPlaceholder.length
-        },
-        '+input'
-      )
+      insertCode(`![](${link})`)
     })
-    .catch(() => {
-      editor.replaceRange(
-        '',
-        cursor,
-        {
-          line: cursor.line,
-          ch: cursor.ch + uploadPlaceholder.length
-        },
-        '+input'
-      )
+    .catch((error) => {
+      console.error('error while uploading file', error)
```

Comment:
Shouldn't this notify the user instead of putting it in the console?

Reply:
Not now. I want change this in another pr.